iothub_client/samples/iotedge_downstream_device_sample/iotedge_downstream_device_sample.c

```diff
@@ -0,0 +1,313 @@
+    {
+        (void)printf("Failure createing Iothub device.  Hint: Check you connection string.\r\n");
+    }
+    else
+    {
+        // Setting message callback to get C2D messages
+        (void)IoTHubDeviceClient_SetMessageCallback(device_handle, receive_msg_callback, NULL);
+        // Setting connection status callback to get indication of connection to iothub
+        (void)IoTHubDeviceClient_SetConnectionStatusCallback(device_handle, connection_status_callback, NULL);
+
+        // Set any option that are neccessary.
+        // For available options please see the iothub_sdk_options.md documentation
+
+        // Turn tracing on/off
+        // bool traceOn = true;
+        // (void)IoTHubDeviceClient_SetOption(iothub_handle, OPTION_LOG_TRACE, &traceOn);
+
+        // Provide the Azure IoT device client with the Edge root
+        // X509 CA certificate that was used to setup the Edge runtime
+        cert_string = obtain_edge_ca_certificate();
+        (void)IoTHubDeviceClient_SetOption(device_handle, OPTION_TRUSTED_CERT, cert_string);
+
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_OVER_WEBSOCKETS
+        //Setting the auto URL Encoder (recommended for MQTT). Please use this option unless
+        //you are URL Encoding inputs yourself.
+        //ONLY valid for use with MQTT
+        //bool urlEncodeOn = true;
+        //(void)IoTHubDeviceClient_SetOption(device_handle, OPTION_AUTO_URL_ENCODE_DECODE, &urlEncodeOn);
+#endif
+
+        if (proxy_host)
+        {
+            HTTP_PROXY_OPTIONS http_proxy_options = { 0 };
+            http_proxy_options.host_address = proxy_host;
+            http_proxy_options.port = proxy_port;
+            http_proxy_options.username = proxy_username;
+            http_proxy_options.password = proxy_password;
+
+            if (IoTHubDeviceClient_SetOption(device_handle, OPTION_HTTP_PROXY, &http_proxy_options) != IOTHUB_CLIENT_OK)
+            {
+                (void)printf("failure to set proxy\n");
+            }
+        }
+
+        for (size_t index = 0; index < MESSAGE_COUNT; index++)
+        {
+            // Construct the iothub message from a string or a byte array
+            message_handle = IoTHubMessage_CreateFromString(telemetry_msg);
+            //message_handle = IoTHubMessage_CreateFromByteArray((const unsigned char*)telemetry_msg, strlen(telemetry_msg)));
+
+            // Set Message property
+            (void)IoTHubMessage_SetMessageId(message_handle, "MSG_ID");
+            (void)IoTHubMessage_SetCorrelationId(message_handle, "CORE_ID");
+            (void)IoTHubMessage_SetContentTypeSystemProperty(message_handle, "application%2fjson");
+            (void)IoTHubMessage_SetContentEncodingSystemProperty(message_handle, "utf-8");
+
+            // Add custom properties to message
+            (void)IoTHubMessage_SetProperty(message_handle, "property_key", "property_value");
+
+            (void)printf("Sending message %d to IoTHub\r\n", (int)(index + 1));
```

Comment:
>(void)printf("Sending message %d to IoTHub\r\n", (int)(index + 1)); [](start = 12, length = 67)

Consider for additional clarity having this being something like "Sending message %d to IoTHub via Edge gateway".  What is written is technically correct too of course.

Reply:
Good suggestion. Fixed.